### ISSUE TITLE
make crab server UI work on K8s

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#6293]
we alsays get:
```
Task info not loaded
```
from crabserver-k8s-prod.cmsweb.cern.ch or cmsweb-testbed.cern.ch (latter one is a K8s FE).

**Comment by belforte** [comment]
quick debugging plan:
- check that this is reproduced on cmsweb-tesbed.
- Then Prajesh, should be able to log into the pod and hack the code to make it somehow print what it is doing and why it fails. You will need to restart the service inside the container every time of course, if you do not have the command at hand, ask me, but I think it is on one twiki page

**Comment by belforte** [comment]
better yet, let's debug on https://cmsweb-test2.cern.ch/crabserver/ui so there's a single pod

**Comment by belforte** [comment]
I tried
https://cmsweb-test2.cern.ch/crabserver/ui/task/201118_213829%3Abfrancis_crab_Higgsino900GeV_ctau10000cm_step1/dbver/prod
while doing a tail -f inside the pod for
/data/srv/logs/crabserver/crabserver-20201215-crabserver-7cbf7476cd-g9t8d.log
and found
```
[15/Dec/2020:11:32:01] crabserver-7cbf7476cd-g9t8d 127.0.0.1 "GET /crabserver/prod/info HTTP/1.1" 404 Not Found [data: 299 in 723 out 1324 us ] [auth: OK "" "" ] [ref: "" "Go-http-client/1.1" ]
[15/Dec/2020:11:32:11]  SERVER REST ERROR WMCore.REST.Error.NoSuchInstance 1d8f03e22d4e7bb3e73b7281ca055ac3 (No such instance)
[15/Dec/2020:11:32:11]    Traceback (most recent call last):
[15/Dec/2020:11:32:11]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/REST/Server.py", line 723, in default
[15/Dec/2020:11:32:11]        return self._call(RESTArgs(list(args), kwargs))
[15/Dec/2020:11:32:11]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/REST/Server.py", line 762, in _call
[15/Dec/2020:11:32:11]        self._precall(param)
[15/Dec/2020:11:32:11]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/REST/Server.py", line 1905, in _precall
[15/Dec/2020:11:32:11]        raise NoSuchInstance()
[15/Dec/2020:11:32:11]    NoSuchInstance: NoSuchInstance 1d8f03e22d4e7bb3e73b7281ca055ac3 [HTTP 404, APP 206, MSG 'No such instance', INFO None, ERR None]
[15/Dec/2020:11:32:11] crabserver-7cbf7476cd-g9t8d 127.0.0.1 "GET /crabserver/prod/info HTTP/1.1" 404 Not Found [data: 299 in 723 out 1576 us ] [auth: OK "" "" ] [ref: "" "Go-http-client/1.1" ]

```

**Comment by belforte** [comment]
Indeed cmsweb-test2 has no access to prod atm
I suggest to use a task in preprod and use cmswen-test2. Having multiple pods makes debugging painful
SOrry if I suggested to use testbed in the first place

**Comment by belforte** [comment]
so this works
https://cmsweb.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod
while this fails
https://cmsweb-test2.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod
with
```
[15/Dec/2020:11:37:47] crabserver-7cbf7476cd-g9t8d 188.184.72.209 "GET /crabserver/preprod/workflow?workflow=201021_090735:ddirmait_crab_20201021_110728 HTTP/1.1" 500 Internal Server Error [data: 4809 in 743 out 30878 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=belforte/CN=373708/CN=Stefano Belforte" "" ] [ref: "https://cmsweb-test2.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0" ]
[15/Dec/2020:11:37:48]  SERVER OTHER ERROR pycurl.error b70dcd4ca4d1dbbf37b2f92a7f72b8a1 ((35, 'error:14094415:SSL routines:SSL3_READ_BYTES:sslv3 alert certificate expired'))
[15/Dec/2020:11:37:48]    Traceback (most recent call last):
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/REST/Server.py", line 723, in default
[15/Dec/2020:11:37:48]        return self._call(RESTArgs(list(args), kwargs))
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/REST/Server.py", line 800, in _call
[15/Dec/2020:11:37:48]        v(apiobj, request.method, api, param, safe)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/CRABInterface/Utils.py", line 171, in wrapped_func
[15/Dec/2020:11:37:48]        args[0].allCMSNames = CMSSitesCache(sites=CRIC().getAllPSNs(), cachetime=mktime(gmtime()))
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/CRIC/CRIC.py", line 124, in getAllPSNs
[15/Dec/2020:11:37:48]        sitenames = self._CRICSiteQuery(callname='site-names')
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/CRIC/CRIC.py", line 86, in _CRICSiteQuery
[15/Dec/2020:11:37:48]        sitenames = self._getResult(uri, callname=callname, args=extraArgs)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/CRIC/CRIC.py", line 59, in _getResult
[15/Dec/2020:11:37:48]        data = self.refreshCache(cachedApi, apiUrl)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/Service.py", line 205, in refreshCache
[15/Dec/2020:11:37:48]        self.getData(cachefile, url, inputdata, incoming_headers, encoder, decoder, verb, contentType)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/Service.py", line 284, in getData
[15/Dec/2020:11:37:48]        contentType=contentType)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 165, in makeRequest
[15/Dec/2020:11:37:48]        result, response = self.makeRequest_pycurl(uri, data, verb, headers)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 183, in makeRequest_pycurl
[15/Dec/2020:11:37:48]        ckey=ckey, cert=cert, capath=capath)
[15/Dec/2020:11:37:48]      File "/data/srv/HG2011d/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/lib/python2.7/site-packages/WMCore/Services/pycurl_manager.py", line 252, in request
[15/Dec/2020:11:37:48]        curl.perform()
[15/Dec/2020:11:37:48]    error: (35, 'error:14094415:SSL routines:SSL3_READ_BYTES:sslv3 alert certificate expired')
[15/Dec/2020:11:37:48] crabserver-7cbf7476cd-g9t8d 188.184.72.209 "GET /crabserver/preprod/workflow?workflow=201021_090735:ddirmait_crab_20201021_110728 HTTP/1.1" 500 Internal Server Error [data: 4809 in 743 out 27471 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=belforte/CN=373708/CN=Stefano Belforte" "" ] [ref: "https://cmsweb-test2.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0" ]

```

**Comment by belforte** [comment]
@sharmaprajesh I let it to you to look into making sure that correct libcurl is used by the crabserver process, it needs to be from our rpm's not from /usr/lib (the one from SL7 lacks SSL support)

**Comment by belforte** [comment]
with reference to directories in the crabserver pod of cmsweb-test2:
@sharmaprajesh the UI page starts from
/data/srv/current/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/data/html/index.html
and if you go to /data/srv/current/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026
you can find all javascript referenced there with
```
[root@crabserver-7cbf7476cd-g9t8d v3.201026]# find . -name '*js'
./data/doc/_static/doctools.js
./data/doc/_static/jquery-1.11.1.js
./data/doc/_static/jquery.js
./data/doc/_static/searchtools.js
./data/doc/_static/underscore-1.3.1.js
./data/doc/_static/underscore.js
./data/doc/_static/websupport.js
./data/doc/searchindex.js
./data/script/gzip.js
./data/script/jquery/jquery.min.js
./data/script/jquery/spin.min.js
./data/script/misc/bootstrap.min.js
./data/script/task_info.js
[root@crabserver-7cbf7476cd-g9t8d v3.201026]# pwd
/data/srv/current/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026
[root@crabserver-7cbf7476cd-g9t8d v3.201026]# 
```
I have NO idea what `./data/doc/_static/` directory is, it is not present in our source code.

**Comment by sharmaprajesh** [comment]
Sorry, I am confused here. Should I submit a task with following configuration?

config.General.instance = ‘other’
config.General.restHost = ‘cmsweb-test2.cern.ch’
config.General.dbInstance = ‘preprod’ 

or something elase?

**Comment by belforte** [comment]
links to code in this repo:
https://github.com/dmwm/CRABServer/blob/master/src/html/index.html
https://github.com/dmwm/CRABServer/blob/master/src/script/gzip.js
https://github.com/dmwm/CRABServer/blob/master/src/script/task_info.js
https://github.com/dmwm/CRABServer/blob/master/src/script/jquery/jquery.min.js
https://github.com/dmwm/CRABServer/blob/master/src/script/jquery/spin.min.js
and various CSS templates are in
https://github.com/dmwm/CRABServer/tree/master/src/css

And when I had to put my hands in this UI code a few years ago, I tried to capture all I learnt in here
https://github.com/dmwm/CRABServer/issues/5690

**Comment by sharmaprajesh** [comment]
Thanks Stefano. I understood it is jquery ( a JS framework) and I know it very well.
But Please let me know the correct configuration which I have to use to submit CRAB Task for this debugging task.

**Comment by belforte** [comment]
Finally, about restarting service inside the pod, picking from `/data/run.sh`:
```
export srv=crabserver
/data/srv/current/config/$srv/manage start 'I did read documentation'
```
or directly
```
/data/srv/current/config/crabserver/manage start 'I did read documentation'
```
of course all of use did read the documentation !!!!!

**Comment by belforte** [comment]
Prajesh, you do not need to submit a new task, any existing record in the DB will do.
My main problem with this in the past was simply `how to add print in the javascript`
I think I managed to make it output some message in html at some point.
But if the problem is really and only a bad version of curl, it should be reasonably easy to fix.
I guess we never payed attention to the fact that the CRABSERVER sort of talks to itself !

**Comment by belforte** [comment]
in other words, you should be able to debug things simply pointing your browser to
https://cmsweb-test2.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod

There still one mistery to me, that that page does not show anything, while cmsweb-testbed or cmsweb-k8s-prod print the  `Task info not loaded`. I do not know why there is a differency. But let's fix it in test2 first, then update testbed and let's see.

**Comment by ddaina** [comment]
> There still one mistery to me, that that page does not show anything, while cmsweb-testbed or cmsweb-k8s-prod print the Task info not loaded. I do not know why there is a differency. But let's fix it in test2 first, then update testbed and let's see.

both cmsweb-testbed and cmsweb-k8s-prod loads properly for me information if i try to check task from preprod. While if I check task from production, I get  `Task info not loaded`. Example task from prod DB: `201215_101708:ddirmait_crab_20201215_111515`

**Comment by belforte** [comment]
oh my... things are not making sense (to me)

**Comment by belforte** [comment]
prajesh, if you need to debug in testbed, i guess you can reduce crabserver to 1 pod using k8s scale command

**Comment by belforte** [comment]
so the error on cmsweb-test2 must be something else...
maybe really an expired cert..

**Comment by ddaina** [comment]
@belforte , so this link https://cmsweb-testbed.cern.ch/crabserver/ui/task/201215_101708%3Addirmait_crab_20201215_111515/ in theory should work, right? It tries to access task info from prod DB. 
Below you can see corresponding log message from `crabserver-20201215-crabserver-778b49d4fc-ck8h5.log` in `vocms0750`:
```
[15/Dec/2020:12:57:06]  RESTSQL:qbKVhwnWUjxu prepare [SELECT tm_taskname, panda_jobset_id, tm_task_status, tm_task_command, tm_user_role, tm_user_group,
        tm_task_failure, tm_split_algo, tm_split_args, panda_resubmitted_jobs, tm_save_logs, tm_username,              tm_user_dn, tm_arguments, tm_input_data
set, tm_dbs_url, tm_task_warnings, tm_publication, tm_user_webdir, tm_asourl,              tm_asodb, tm_output_dataset, tm_collector, tm_schedd, tm_dry_run, c
lusterid, tm_start_time, tw_name              FROM tasks WHERE tm_taskname=:taskname]
[15/Dec/2020:12:57:06]  RESTSQL:qbKVhwnWUjxu execute: () {'taskname': '201215_101708:ddirmait_crab_20201215_111515'}
[15/Dec/2020:12:57:06]  SERVER REST ERROR WMCore.REST.Error.ExecutionError 60f948f8463af5e9fb0dfe9161875e5f (Execution error)
[15/Dec/2020:12:57:06]    Traceback (most recent call last):
[15/Dec/2020:12:57:06]      File "/data/srv/HG2101a/sw/slc7_amd64_gcc630/cms/crabserver/v3.201027-comp3/lib/python2.7/site-packages/WMCore/REST/Format.py", li
ne 249, in stream_chunked
[15/Dec/2020:12:57:06]        % (obj, type(obj), str(exp)))
[15/Dec/2020:12:57:06]    UnboundLocalError: local variable 'obj' referenced before assignment
[15/Dec/2020:12:57:06]  SERVER HTTP ERROR cherrypy._cperror.HTTPError 88d77a346438b055f4952903e56ca8d0 ((500, 'Execution error'))
[15/Dec/2020:12:57:06]    Traceback (most recent call last):
[15/Dec/2020:12:57:06]      File "/data/srv/HG2101a/sw/slc7_amd64_gcc630/cms/crabserver/v3.201027-comp3/lib/python2.7/site-packages/WMCore/REST/Server.py", li
ne 723, in default
[15/Dec/2020:12:57:06]        return self._call(RESTArgs(list(args), kwargs))
[15/Dec/2020:12:57:06]      File "/data/srv/HG2101a/sw/slc7_amd64_gcc630/cms/crabserver/v3.201027-comp3/lib/python2.7/site-packages/WMCore/REST/Server.py", li
ne 839, in _call
[15/Dec/2020:12:57:06]        return stream_maybe_etag(apiobj.get('etag_limit', self.etag_limit), etagger, reply)
[15/Dec/2020:12:57:06]      File "/data/srv/HG2101a/sw/slc7_amd64_gcc630/cms/crabserver/v3.201027-comp3/lib/python2.7/site-packages/WMCore/REST/Format.py", li
ne 590, in stream_maybe_etag
[15/Dec/2020:12:57:06]        raise cherrypy.HTTPError(int(err), message)
[15/Dec/2020:12:57:06]    HTTPError: (500, 'Execution error')
[15/Dec/2020:12:57:06] crabserver-778b49d4fc-ck8h5 84.46.229.85 "GET /crabserver/preprod/task?subresource=webdirprx&workflow=201215_101708:ddirmait_crab_20201215_111515 HTTP/1.1" 500 Internal Server Error [data: 4538 in 746 out 618996 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=ddirmait/CN=803460/CN=Daina Dirmaite" "" ] [ref: "https://cmsweb-testbed.cern.ch/crabserver/ui/task/201215_101708%3Addirmait_crab_20201215_111515/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0" ]
```

**Comment by belforte** [comment]
is that task in prod or preprod ?

**Comment by belforte** [comment]
last line in log that you pasted points to preprod

**Comment by belforte** [comment]
should try same url with /dbver/prod added at right and see which error you get

**Comment by belforte** [comment]
and really, set testbed to only one crabserver pod, limit number of log files

**Comment by ddaina** [comment]
you are right, I overlooked that last line points to preprod. 

Ok, I am scaling down number of pods.

**Comment by belforte** [comment]
maybe the UI internals does not adapt correctly to prod vs preprod

**Comment by ddaina** [comment]
done, only one pod is running now in k8s-testbed

**Comment by ddaina** [comment]
so, if i try to access https://cmsweb-testbed.cern.ch/crabserver/ui/task/201215_101708%3Addirmait_crab_20201215_111515/dbver/prod , there is nothing in the log file about it.

**Comment by ddaina** [comment]
i was wrong, there is this line:
```
[15/Dec/2020:15:13:20] crabserver-778b49d4fc-t5jx2 84.46.229.85 "GET /crabserver/ui/task/favicon.ico HTTP/1.1" 200 OK [data: 4389 in 2527 out 1372 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=ddirmait/CN=803460/CN=Daina Dirmaite" "" ] [ref: "https://cmsweb-testbed.cern.ch/crabserver/ui/task/201215_101708%3Addirmait_crab_20201215_111515/dbver/prod" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0" ]
```

**Comment by belforte** [comment]
the annoying thing is that brwser or served i never figured out which, caches this page for a LONG time. 
even if yoy reload even if yoy reload bypassing browser cache

**Comment by belforte** [comment]
i did not find a way to remove this, it is bad also if you look at jo status or transfer 
need at least to restart brwser

**Comment by belforte** [comment]
otoh the url which you send is for UI, the line in the log which points to preprod and guves the error is created by the jquery script

**Comment by belforte** [comment]
somehow it looks like this is not working and points to preprod even when it should not
https://github.com/dmwm/CRABServer/blob/9f26e3a5249398bcc593e5ee2a569921fd55db6b/src/script/task_info.js#L771-L797
I defer to Prajesh to add diagnostics to this thing.

**Comment by belforte** [comment]
the fact that it wokrs on theVM cluster rules out simple bugs, that switch usually works finely

**Comment by belforte** [comment]
I think one can add proper lines in jsquery to make it output messages in html . Then of coure there's the business of bypassing the stupid cache, at worst one restarts the service (i.e.http server) every time.
so, just add printouts... the old way.

**Comment by sharmaprajesh** [comment]
I updated this file in test cluster[1] and now this url popup "dev" https://cmsweb-test2.cern.ch/crabserver/ui
@belforte Is this expected output.

[1]
771     function setUrls(dbVersion) {
772         alert(dbVersion);   <--------------
773
in
[root@crabserver-7cbf7476cd-g9t8d data]# vi ./script/task_info.js
[root@crabserver-7cbf7476cd-g9t8d data]# pwd
/data/srv/current/sw/slc7_amd64_gcc630/cms/crabserver/v3.201026/data

**Comment by belforte** [comment]
dev as default is ok, but then it should be possible to change in the pull down menu. we can change the code to make it default to prod, of course

**Comment by belforte** [comment]
indeed after i change in the pull down menu it says "prod" but still task i not loaded

**Comment by belforte** [comment]
andnow the page has this, cool !
```
Hello
dbVersion=prod
dbVersion=prod
dbVersion=prod
```

**Comment by sharmaprajesh** [comment]
Now, We can show any value on the main page with the following code[1]. Replace dbVersion variable with your choice.
We can put this line in any java script code.
I am trying to understand the code.

[1]
document.getElementById('output').innerHTML += 'dbVersion='+dbVersion+'<br/>';

**Comment by sharmaprajesh** [comment]
` document.getElementById('output').innerHTML += 'dbVersion='+dbVersion+'<br/>';`

**Comment by belforte** [comment]
thanks Prajesh, I think you shoud print what happens around line 202
somehow the taskInfoUrl appears not the right one

**Comment by belforte** [comment]
by the way test2 is not configured to be able to access prod DB. But I have no problem if you hack the testbed pod. we are the only ones using it

**Comment by belforte** [comment]
is this just a tip, or did you change the script to make it possible ?
if the latter, make sure you port the change to GH, thanks !
If the former, let's add to the tricks twiki.

> Now, We can show any value on the main page with the following code[1]. Replace dbVersion variable with your choice.
> We can put this line in any java script code.
> I am trying to understand the code.
> 
> [1]
> document.getElementById('output').innerHTML += 'dbVersion='+dbVersion+'
> ';

**Comment by sharmaprajesh** [comment]
I select this task[1], and with different drop down i.e. prod , preprod and dev, I got output.[2]

@belforte , Could you please help me to understand from here that What exactly the problem we are facing and trying to resolve.
 
[1]
https://cmsweb-test2.cern.ch/crabserver/ui/task/200722_165928%3Arvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B

[2]
dbVersion=prod
taskInfoUrl=https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B

dbVersion=preprod
taskInfoUrl=https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B

dbVersion=dev
taskInfoUrl=https://cmsweb-test2.cern.ch/crabserver/dev/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb-test2.cern.ch/crabserver/dev/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B

**Comment by sharmaprajesh** [comment]
`dbVersion=prod
taskInfoUrl=https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
dbVersion=preprod
taskInfoUrl=https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
dbVersion=dev
taskInfoUrl=https://cmsweb-test2.cern.ch/crabserver/dev/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb-test2.cern.ch/crabserver/dev/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
`

**Comment by sharmaprajesh** [comment]
"Task info not loaded" in all cases.

**Comment by sharmaprajesh** [comment]
I changed the script in test2 cluster, So that you may also test this.

**Comment by belforte** [comment]
Thoseook correct Sent from. Stefano's phoneOn Dec 15, 2020 17:26, sharmaprajesh <notifications@github.com> wrote:
I select this task[1], and with different drop down i.e. prod , preprod and dev, I got output.[2]
@belforte , Could you please help me to understand from here that What exactly the problem we are facing and trying to resolve.
[1]
https://cmsweb-test2.cern.ch/crabserver/ui/task/200722_165928%3Arvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
[2]
dbVersion=prod
taskInfoUrl=https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
dbVersion=preprod
taskInfoUrl=https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
dbVersion=dev
taskInfoUrl=https://cmsweb-test2.cern.ch/crabserver/dev/task?subresource=search&workflow=
inputTaskName=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B
url=https://cmsweb-test2.cern.ch/crabserver/dev/task?subresource=search&workflow=200722_165928:rvaladao_crab_LLGDV_Analysis_DATA_MET_RUN2017B

**Comment by belforte** [comment]
So need to add more diagnostics downstream of thatSent from. Stefano's phoneOn Dec 15, 2020 17:40, sharmaprajesh <notifications@github.com> wrote:
"Task info not loaded" in all cases.

**Comment by belforte** [comment]
I am getting VERY confused here.

This works:
https://cmsweb-testbed.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod
and this works too:
https://cmsweb-k8s-prod.cern.ch/crabserver/ui/task/201021_090735%3Addirmait_crab_20201021_110728/dbver/preprod

**Comment by belforte** [comment]
I start to suspect that problem is with proxied webdir url. Maybe it is not build correctly, maybe redirection rules are missing on K8s FEs.

**Comment by belforte** [comment]
notice. proxywebdir url is obtained via queries like this
https://cmsweb-test2.cern.ch/crabserver/preprod/task?subresource=webdirprx&workflow=201021_090735:ddirmait_crab_20201021_110728
which returns
```
{"result": [
 "https://cmsweb.cern.ch/scheddmon/0199/cms349/201021_090735:ddirmait_crab_20201021_110728"
]}
```
so I still do not see what's wrong in test2 and even what I wrote earlier can be an illusion due to extensive caching here. At times even restarting crabserver is not enough to see new code in action. Of course browser caches jscript client side for performance... and all that stuff.

**Comment by belforte** [comment]
hmmm...... I decided to take seriously the pycurl error here https://github.com/dmwm/CRABServer/issues/6293#issuecomment-745204199
and commented out the line which caused the traceback. Et voila':
![Screenshot from 2020-12-15 21-33-39](https://user-images.githubusercontent.com/1837785/102269762-b7ac8580-3ee2-11eb-97ba-b87be500c6b7.png)

**Comment by belforte** [comment]
OTOH, while it is possible that cmsweb-test2 has odd problems which cause it to fail to talk to CRIC (the root of that traceback), I would have thought that it works on other cluster or we should have seen more problems.
OTOH... maybe this only affects this UI and some specific submission configuration that we never tested on k8s ?

**Comment by belforte** [comment]
in a way.. the problem was showing up on test2 as an empy page, not with the "task not loaded" message.
the search for CRIC string returns nothing in crabserver logs on vocms0750:/cephfs/testbed/crab-logs

Maybe we should switch to look into that. Anyhow I will leave test2 as is now, with all debug printout in task_info.js commented out and this hack in  lib/python2.7/site-packages/CRABInterface/Utils.py
```
                args[0].allCMSNames = []#CMSSitesCache(sites=CRIC().getAllPSNs(), cachetime=mktime(gmtime()))
                args[0].allPNNNames = []#CMSSitesCache(sites=CRIC().getAllPhEDExNodeNames(), cachetime=mktime(gmtime()))

```
the hack is the insertion 0f `[]#`

**Comment by belforte** [comment]
you are more than welcome to try test2 with other tasks, of course in preprod !

**Comment by belforte** [comment]
I did some tests on cmsweb-testbed.,
I zeroed in on this code fragment (see the alert(); which I inserted):
```
    /**
     * Called on task info search form submission. It then fetches JSON data
     * and inserts it into the #task-info-table.
     */
    function displayTaskInfo(errHandler) {
        var xmlhttp = new XMLHttpRequest();
        // var url = "https://mmascher-mon.cern.ch/crabserver/dev/task?subresource=search&workflow=";
        var url = taskInfoUrl + inputTaskName;
        alert('in displayTaskInfo. Url is:');
        alert(url);

        function queryApi(url) {
            alert('inside the query. url is '+url);
            $.ajax(url)
                .done(function(data) {
                    // Storing the data for the use of other display functions
                    alert('query is .done');
                    taskInfo = data;

                    // Creating table contents
                    for (i = 0; i < data.desc.columns.length; i++) {
                        $("#task-info-table tbody")
                            .append("<tr><td>" + data.desc.columns[i] + "</td><td>" + data.result[i] + "</td></tr>");
                    }
                    loadOtherData();

                })
                .fail(function(xhr) {
                    alert('query is .fail');
                    var headers = xhr.getAllResponseHeaders().toLowerCase();
                    alert(headers);
                    errHandler(new ServerError(headers));
                    alert('calling loadOtherData');
                    loadOtherData();
                })
        };
        queryApi(url);
```
the URL shown by `alert('inside the query. url is '+url);` is correct
https://cmsweb.cern.ch/crabserver/prod/task?subresource=search&workflow=201215_101708:ddirmait_crab_20201215_111515

I can copy/paste it from the popup box into the browser and it works, but then code falls to `.fail` and headers appear empty.
I am out of ideas.
unless.... maybe something prevents K8s BE's to talk to cmsweb VM cluster ?

**Comment by belforte** [comment]
my code was lost in the crabserver restart... see https://mattermost.web.cern.ch/cms-o-and-c/pl/z1t618u38jbypeprkn1xy5936e

I did a quick test to replace cmsweb.cern.ch with cmsweb-testbed.cern.ch in the URL defintions in
https://github.com/dmwm/CRABServer/blob/9f26e3a5249398bcc593e5ee2a569921fd55db6b/src/script/task_info.js#L773-L779
and looks like it works.
calling it a day.

A better solution using domain.name like done below for preprod etc. will be better. We will need to test and then ask Imran for a re-deployment in production.

There are other places where that script containsw the string `cmsweb.cern.ch` which may or may not matter.

On the bright side the problem with CRIC seems limited to test2, maybe it really has an expired certificate whjich is used only in this case !

I hate to have calls to CRIC from inside CRABServer, and put some time trying to get rid of it, but it was not simple and I gave up (it works, don't fix it). Given enough time, could be surely revisited.
